src/ProjectConfiguration.ts

```diff
@@ -254,6 +261,40 @@ export function normalizeProjectConfig(
     );
   }
 
+  /**
+   * FbOS 3.0 has a max size limit on the app manifest of 1K
+   * so if a developer specifies language + locale variants (eg en and en-US)
+   * for all languages they'll quickly exceed this. This code merges the language
+   * tag into a locale tag unless one is present.
+   */
+  const culledLocales: string[] = [];
+  mergedConfig.i18n = lodash.mapKeys(mergedConfig.i18n, (_, locale) => {
+    const localeMapping: { [locale: string]: string | undefined } = {
+      en: 'en-US',
+      de: 'de-DE',
+      it: 'it-IT',
+      es: 'es-ES',
+      fr: 'fr-FR',
+      ja: 'ja-JP',
+      ko: 'ko-KR',
+      nl: 'nl-NL',
+      sv: 'sv-SE',
+    };
+    const mappedLocale = localeMapping[locale];
+
+    if (mappedLocale === undefined) {
+      return normalizeLanguageTag(locale);
+    }
+
+    if (mergedConfig.i18n[mappedLocale] !== undefined) {
+      culledLocales.push(locale);
+      return locale;
+    }
+
+    return mappedLocale;
```

Comment:
i think i might be missing something; in this case, if we have `en` and `en-US` keys in this order in `mergedConfig.i18n`, the `en` object will be mapped to the `en-US` key, and after that `en-US` key comes and overrride the previous object (set by mapping `en`). From what we discussed we have different data on `en` and `en-US` and we need both.

e.g.

const dict = {
  "key1": {
    "ala": "lol"
  },
  "key2": {
    "bala": "lol2"
  }
};


result = _.mapKeys(dict, function(value, key) {
  if(key === "key1") {
    return "key2";
  }
  return key;
});

Reply:
Think we talked about this in person, but for history: `en` should not replace `en-US`. The `en` value is just ignored in this instance. We don't warn developers because this whole thing is just a workaround and we will likely use `en` info in future.